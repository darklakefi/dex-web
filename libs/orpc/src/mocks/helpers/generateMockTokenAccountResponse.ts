import { randNumber, randUuid, seed } from "@ngneat/falso";
import { type DAS, Interface, OwnershipModel } from "helius-sdk";
import { times } from "remeda";
import { FIXED_SEED } from "./constants";

export function generateTokenAccountsResponseList({
  count = 10,
}: {
  count?: number;
} = {}): DAS.GetTokenAccountsResponse {
  seed(FIXED_SEED);
  return {
    limit: 10,
    token_accounts: times(count, (_i) => ({
      address: randUuid(),
      amount: randNumber({ max: 1000000, min: 1 }),
      id: randUuid(),
      interface: Interface.V1NFT,
      mint: randUuid(),
      mutable: true,
      ownership: {
        delegate: undefined,
        delegated: false,
        frozen: false,
        owner: randUuid(),
        ownership_model: OwnershipModel.SINGLE,
      },
      token_extensions: [],
    })),
    total: 10,
  };
}
