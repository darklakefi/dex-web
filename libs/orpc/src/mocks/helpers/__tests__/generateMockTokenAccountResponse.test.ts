import { generateTokenAccountsResponseList } from "../generateMockTokenAccountResponse";

describe("generateTokenAccountsResponseList", () => {
  it("should generate a list of token accounts", () => {
    const response = generateTokenAccountsResponseList();

    expect(response).toMatchSnapshot();
    expect(response.token_accounts).toBeDefined();
    expect(response.token_accounts?.length).toBe(10);
  });
});
