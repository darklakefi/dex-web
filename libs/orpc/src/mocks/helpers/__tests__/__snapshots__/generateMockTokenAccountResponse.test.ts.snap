// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateTokenAccountsResponseList > should generate a list of token accounts 1`] = `
{
  "limit": 10,
  "token_accounts": [
    {
      "address": "3c4dfb0c-f84d-41ea-ac6d-be433685d544",
      "amount": 906466,
      "id": "281d9e83-4d8e-48e5-ac10-6dad7426e481",
      "interface": "V1_NFT",
      "mint": "17f4a3c0-badc-4a17-8f21-89174526ced4",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "7a41ef27-d027-41fc-bd02-d516d3adb894",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "4cb6932d-b540-4464-b66e-b6964d4e74c9",
      "amount": 290423,
      "id": "f8db0d72-7056-4658-8b56-c9dd3fc4fd8f",
      "interface": "V1_NFT",
      "mint": "4bd821d2-a605-4715-b3ca-0f1a2f725144",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "a62dbbda-abef-4b73-a50e-6910dee76cbd",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "f688dd29-19c4-4be5-b1e9-096ec3e5963c",
      "amount": 498798,
      "id": "dfc9b8ec-b396-42b2-b1f3-3a4409e78b44",
      "interface": "V1_NFT",
      "mint": "32821048-e405-4c14-a468-15a5dd445f9f",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "b722ac27-7f13-4b0b-a005-8186974372c6",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "c78e650c-793c-43f9-ad3e-7e59d645c6c2",
      "amount": 741975,
      "id": "f0c08392-0a55-49e0-a12e-b620e9565587",
      "interface": "V1_NFT",
      "mint": "352c0bb8-ab6d-4f46-8dd8-554d99d4a1a1",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "bf125007-6b50-4d9e-9078-53955834d6be",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "653cfe42-e62a-4d68-8c64-f507e3384624",
      "amount": 599623,
      "id": "f29068e1-dd6c-40c6-b89a-30eb2cfd1a73",
      "interface": "V1_NFT",
      "mint": "0a4ebaa4-97a4-4fc8-a90e-70ae24c17ece",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "0b7e2aad-637c-4893-a275-0537d9a1e4b6",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "96092c68-cb6d-40f2-9cbe-e2b393adfc17",
      "amount": 418322,
      "id": "b6e307ab-05a2-4746-b3ba-ae47e2ddef75",
      "interface": "V1_NFT",
      "mint": "5b85dafd-df63-4696-8b02-5ed0ca7bbed7",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "5243090a-8438-4ecd-9853-ccd8999287df",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "0b110e8f-87b9-4990-8de4-0ec34c74a091",
      "amount": 155566,
      "id": "6f0cc9ee-bad6-45d1-9e02-34da2b46bbab",
      "interface": "V1_NFT",
      "mint": "b4622e48-1afa-44ae-be94-d19be7b39336",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "0d24d56d-8f4e-435d-9604-18ef175b5762",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "0e82c196-ce91-4fb4-86dd-be60ac6bce66",
      "amount": 525454,
      "id": "7535f17a-ebe6-475c-aca9-08e12658591f",
      "interface": "V1_NFT",
      "mint": "85645e43-bd3f-49df-885d-8eee9e89d595",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "04ae405b-b228-438d-882f-68277744a612",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "71ff1785-8ddb-4314-bb89-f4ecf04cf20e",
      "amount": 612304,
      "id": "16c8ed07-c068-4385-bec7-ab89ceb24926",
      "interface": "V1_NFT",
      "mint": "9bac24ee-d28d-4fd1-9727-ff524106e523",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "ec3f761d-3240-4039-94fc-d8082f8199c2",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
    {
      "address": "c82d3f3c-4a40-45ac-8da9-76c09db29be3",
      "amount": 4038,
      "id": "08614ba1-79c0-4121-ba20-7a3b9d17c15e",
      "interface": "V1_NFT",
      "mint": "ac1cd6bc-f3e1-4ab7-82ba-2d25cf63e63d",
      "mutable": true,
      "ownership": {
        "delegate": undefined,
        "delegated": false,
        "frozen": false,
        "owner": "de4f6677-dd2a-478e-9ed7-c08bd0ce96e2",
        "ownership_model": "single",
      },
      "token_extensions": [],
    },
  ],
  "total": 10,
}
`;
